internal/prometheusbp/spectest/spec.go

```diff
@@ -0,0 +1,203 @@
+package spectest
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+	"testing"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/testutil"
+
+	"github.com/reddit/baseplate.go/errorsbp"
+)
+
+const (
+	client = "client"
+	server = "server"
+)
+
+var (
+	errPrefix         = errors.New("the prefix is not at the beginning of the metric name")
+	errClientServer   = fmt.Errorf("missing %q or %q as the second part of the metric name", client, server)
+	errLength         = errors.New("metric name should have a minimum of three parts")
+	errNotFound       = errors.New("metric not found")
+	errLabelNotFound  = errors.New("label not found")
+	errPrometheusLint = errors.New("prometheus GatherAndLint")
+)
+
+// ValidateSpec validates that the Prometheus metrics being exposed from baseplate.go
+// conform to the baseplate spec.
+func ValidateSpec(t *testing.T, metricPrefix string, wantMetricCount int) {
+	t.Helper()
+
+	var batch errorsbp.Batch
+	batch.AddPrefix("validate spec", validateSpec(metricPrefix))
+
+	if len(batch.GetErrors()) > 0 {
+		t.Errorf(batch.Compile().Error())
+	}
+}
+
+func validateSpec(prefix string) errorsbp.Batch {
+	var batch errorsbp.Batch
+
+	metricFam, err := prometheus.DefaultGatherer.Gather()
+	if err != nil {
+		return batch
+	}
+
+	allMetricNames := buildMetricNames(prefix)
```

Comment:
🔕 since this represents a "shrinking" list of metrics, you could call it `remainingMetricNames` or something -- and then the loop at the end reads super nicely

Reply:
how about `metricNamesUnderTest`?